client/blocks/installments-plan-picker/index.jsx

```diff
@@ -0,0 +1,93 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import React from 'react';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import FormLabel from 'components/forms/form-label';
+import FormRadio from 'components/forms/form-radio';
+
+export class InstallmentsPlanPicker extends React.Component {
+	static propTypes = {
+		initialValue: PropTypes.number,
+		plans: PropTypes.array.isRequired,
+		onChange: PropTypes.func,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onChange: () => null,
+	};
+
+	state = {
+		installmentsInputValue: this.props.initialValue,
+	};
+
+	formatPlans() {
+		return this.props.plans.map( plan => {
+			plan.label = plan.value + ' x ' + plan.payment;
+			return plan;
+		} );
+	}
+
+	handleInstallmentsInput = event => {
+		const installments = parseInt( event.target.value );
+
+		this.setState( {
+			userChangedInstallments: true,
+			installmentsInputValue: installments,
+		} );
+
+		this.props.onChange( installments );
+
+		this.setState( {
```

Comment:
It's possible that you'll run into some race conditions due to `this.setState` being asynchronous. 

I don't see where you're using `this.state.userChangedInstallments`, but if you need to rely on it later perhaps you could set it _after_ `onChange` has done its work? 

Maybe it's a case for the new lifecycle method `getDerivedStateFromProps`? https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops